frontend/src/lib/hooks/useScroll.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from "react";
```

Comment:
I suggests to use @reduxjs/toolkit and slice. Scroll will be reusable if needed.

Reply:
Thank you for the suggestion. I'd like to understand it more clearly before proceeding. So, if I understand correctly, you're proposing storing the landing page scroll position globally for easy access to this state in potential future landing page components? Just to mention, we already utilize react-intersection-observer in every landing page section. Additionally, if I integrate Redux into the useScroll hook, wouldn't it potentially lose its reusability, especially if we have multiple components requiring different scroll positions?

Reply:
This change shouldn't have any effect to scroll behavior because we will simply move local state to global. I was more interested in using scroll state in future if needed.

Reply:
I'm considering a potential issue with this change. Let's say I scroll down on the landing page, and the global state saves this scroll position. If I then navigate to another component, such as the login page, and later return to the landing page by clicking on the logo ("/"), the landing page might incorrectly interpret the saved scroll position as 'scrolled,' even though it's a new session or a different context. Managing the global scroll position introduces maintenance overhead. We'll need to remember to reset it in certain cases when components mount to ensure accurate behavior. If this isn't a serious issue or if I'm mistaken in my thought process, I'm willing to implement the change, but I wanted to raise these considerations for discussion.

Reply:
I didn't take as a consideration. In this case is worth to try and see result. I will left decision for you.

Reply:
OK, if you still think it's profitable, I'm implementing it (but I need some time to remember how to setup redux).